### PR TITLE
Communication for VOF conservation

### PR DESCRIPTION
**Pull request by mbkuhn**
PR implements communication between stages of the VOF transport, fixing previous conservation issues. Also fixes similar issues upon initialization and filtering of the velocity field. Finally, adds a convenient user input for the removal of VOF debris and provides better verbose MultiPhase outputs concerning conservation.

### CODE REVIEW

**Comment by mbkuhn** [comment]
PR will likely cause diffs for the multiphase cases - this is because it's fixing something there.

File: unit_tests/multiphase/test_vof_cons.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                }
+                // Check conservation
+                EXPECT_NEAR(mphase.volume_fraction_sum(), sum_vof0, tol);
+            }
+        }
+
+        if (dir >= 0) {
+            run_algorithm(vof, [&](const int lev, const amrex::MFIter& mfi) {
+                const auto& vof_arr = vof(lev).const_array(mfi);
+                const auto& bx = mfi.validbox();
```
Comment (gdeskos) [inline unit_tests/multiphase/test_vof_cons.cpp:187]: remove this line

File: unit_tests/multiphase/test_vof_cons.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+                // Loop manually through cells to check values
+                for (int i = 0; i < 3; ++i) {
+                    for (int j = 0; j < 3; ++j) {
+                        for (int k = 0; k < 3; ++k) {
+
+                            int icheck;
```
Comment (gdeskos) [inline unit_tests/multiphase/test_vof_cons.cpp:194]: Can you initialize `icheck` to zero ?

**Review by gdeskos — APPROVED** [review]

File: unit_tests/multiphase/test_vof_cons.cpp [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            for (auto& seqn : pde_mgr.scalar_eqns()) {
+                if (seqn->fields().field.base_name() == "vof") {
+                    seqn->initialize();
+                    seqn->compute_advection_term(amr_wind::FieldState::Old);
+                    seqn->post_solve_actions();
+                }
+                // Check conservation
+                EXPECT_NEAR(mphase.volume_fraction_sum(), sum_vof0, tol);
+            }
+        }
```
Comment (sayerhs) [inline unit_tests/multiphase/test_vof_cons.cpp:182]: You don't need the loop in L173. You should be able to access the VOF equation directly from the `pde_mgr` object  before the for loop in L171. 

```C++
auto seqn = pde_mgr<amr_wind::pde::VOF>(amr_wind::pde::VOF::pde_name());
```

Why is the `seqn->initialize()` called within each iteration?

**Review by gdeskos — APPROVED** [review]

**Pull request merged**